src/runtime/registry.cc

```diff
@@ -47,12 +47,27 @@ struct Registry::Manager {
 
   Manager() {}
 
+#ifdef C_REGISTRY
+  ~Manager() {
+    for (auto& it : fmap)
+    {
+      delete it.second;
+      it.second = nullptr;
+    }
+  }
+#endif
+
   static Manager* Global() {
+#ifdef C_REGISTRY
+    static Manager inst;
```

Comment:
This won't work for multiple models unless we are explicitly clearing this object and reusing it, correct?
I think a run with thread sanitizer turned on should confirm this. Can you please attach this report? Thanks.

Reply:
Could you elaborate on why this won't work for multiple models? Actually tvm original implementation prior to their on purpose leakage is identical to what I have here: https://github.com/apache/tvm/pull/2046/commits/f2f7b974744052fe80f2bbe94b6516397fd3b506

Reply:
TVM impl does not handle multiple instances of the model. Manager instance will need to be reused once a model has been released (since this is a `static`). Depending on how much of the BSS segment we use for global objects (ie., TLS size)

Reply:
Address sanitizer should catch this?

Reply:
Attached log from 
1. address sanitizer enabled loading 2 x Resnet18 model [address_sanitizer_fix.log](https://github.com/neo-ai/tvm/files/5884239/address_sanitizer_fix.log)
2. valgrind log with libdlr.so compiled with this PR [original_valgrind.log](https://github.com/neo-ai/tvm/files/5884241/original_valgrind.log)
3. valgrind log with original libdlr.so without this PR [fix_valgrind.log](https://github.com/neo-ai/tvm/files/5884242/fix_valgrind.log)

From the log I don't see problem with loading multiple models for now. Also leak issue with `tvm::runtime::Registry::Register` was resolved. I have logs for loading it for more than 2 times, it is just way too long and basically tell the same story.

Below is the code I use to produce above logs

```
#include <iostream>
#include <neo/runtime/model.hh>

int main()
{
    std::string modelPath = "/home/ubuntu/distribution/resources/models/linux-x64/cpu/resnet18_v1";

    for (int i = 0; i < 1; i++)
    {
        std::cout << "=========== Load model 1 ===========" << std::endl;
        auto m0 = NeoModel(modelPath);
        std::cout << "=========== Load model 2 ===========" << std::endl;
        auto m1 = NeoModel(modelPath);
    }
}
```

Reply:
Makes sense. 
Something to note: It's possible the compiler might optimize this code away since `m0` and `m1` are note used.
Just make sure you  compiled with no optimizations.

Reply:
Have tested again with optimization flags off. Got the same results.